lib/token_scanner.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'jwt'
+require 'base64'
+
+# TokenScanner scans configmaps for JWT-style tokens and logs approaching or expired number of days
+class TokenScanner
+  # warn threshold in days
+  WARN_THRESHOLD = 30
+
+  # Regex pattern to match JWT like strings:
+  #   with non-capturing group, capture any word character (\w) and hyphens followed by . exactly 2 times
+  #   then, capture any word character (\w) and hyphens until end of line
+  #   like: xxx.xxx.xxx, -xx.x-x-x.xx-, etc...
+  JWT_REGEX = /^(?:[\w-]*\.){2}[\w-]*$/.freeze
+
+  def initialize
+    @tokens = {
+      production: [],
+      staging: []
+    }
+  end
+
+  # run runs the scan and logs any existing results existing with an error
+  def run
+    @tokens.each do |context, results|
+      scan(context, results)
+    end
+
+    return unless @tokens[:production].any? || @tokens[:staging].any?
+
+    # log if any tokens are near (or past) expiration
+    log
```

Comment:
Do we want to be more declarative with this method name? Ideally, the name should give us enough info to not need to comment on its purpose. Maybe something even minimal like `log_expirations`